### PR TITLE
[LETS-380] correct make_unique char array

### PR DESCRIPTION
**Pull request by cristiarg**
http://jira.cubrid.org/browse/LETS-380

Regression.
Went under the radar during a recent commit.

### CODE REVIEW

File: src/transaction/log_page_buffer.c [hunk 1]
```diff
@@ -2250,7 +2250,7 @@ logpb_read_page_from_file_or_page_server (THREAD_ENTRY * thread_p, LOG_PAGEID pa
 	{
 	  // context 2)
 	  // *INDENT-OFF*
-	  auto log_page_buffer_uptr = std::make_unique<char> (IO_MAX_PAGE_SIZE);
+	  auto log_page_buffer_uptr = std::make_unique<char []> (IO_MAX_PAGE_SIZE);
```
Comment (razvanradulescu) [inline src/transaction/log_page_buffer.c]: Nit :
Use db_log_page_size () or LOG_PAGESIZE

File: src/storage/page_buffer.c [hunk 2]
```diff
@@ -8140,7 +8140,7 @@ pgbuf_read_page_from_file_or_page_server (THREAD_ENTRY * thread_p, const VPID *
       if (read_from_local)
 	{
 	  // *INDENT-OFF*
-	  auto buffer_uptr = std::make_unique<char> (db_io_page_size ());
+	  auto buffer_uptr = std::make_unique<char []> (db_io_page_size ());
```
Comment (razvanradulescu) [inline src/storage/page_buffer.c]: This style with auto looks dangerous:
According to form (2), if return type of db_io_page_size is size_t, then you would get the behavior expected from previous version.
What if we change the signature of db_io_page_size () in the future ?

https://docs.w3cub.com/cpp/memory/unique_ptr/make_unique

**Review by razvanradulescu — APPROVED** [review]

File: src/storage/page_buffer.c (hunk 2 above)
Comment (cristiarg) [reply src/storage/page_buffer.c]: You meant:
```
According to form (2), if return type of db_io_page_size is _NOT_ size_t, then you would get the behavior expected from previous version.
```
You are correct.

Comment (cristiarg) [reply src/storage/page_buffer.c]: Changed the code to be more explicit.

File: src/transaction/log_page_buffer.c (hunk 1 above)
Comment (cristiarg) [reply src/transaction/log_page_buffer.c]: Done. Changed the code to be more explicit.

**Review by razvanradulescu — APPROVED** [review]

**Pull request merged**